### PR TITLE
[PD-1763] Added models to track outreach results.

### PR DESCRIPTION
**Pull request by aspidites**
Realized that the one big difference between Partner.library and these models is that multiple partners, contacts, and communication records can be made from a single non-user outreach email.

### CODE REVIEW

File: mypartners/models.py [hunk 1]
```diff
@@ -900,7 +900,23 @@ class OutreachWorkflowState(models.Model):
...
-    from_email = models.EmailField(max_length=255, verbose_name="Email", 
-                                   help_text="Email outreach effort sent from.")
+    from_email = models.EmailField(
+        max_length=255, verbose_name="Email",
+        help_text="Email outreach effort sent from.")
     email_body = models.TextField()
     current_workflow_state = models.ForeignKey("OutreachWorkflowState")
+
+
+class OutreachPartner(models.Model):
```
Comment (koonsies) [inline mypartners/models.py]: I'm a bit confused. What makes this better than just having a standard many to many relationship between Outreach and Partner?

Comment (haliscooper) [inline mypartners/models.py]: Came out of a discussion that I remember Jason M saying something and me going, "Oh yeah, those should be separate table" but can't remember why.  Could very well be a column on partner, contacts and contact record that is null 99% of the time, inefficient to index it.  Jason is searching his memory...

Comment (aspidites) [inline mypartners/models.py]: If I understand correctly, this was addressed and I'm keeping this as-is? @JLMcLaugh?

Comment (JLMcLaugh) [inline mypartners/models.py]: There are some comments on the ticket now. @koonsies is right though. We don't need separate models if we're just trying to get a many-to-many relationship.

**Comment by aspidites** [comment]
Now that the dust has settled, I updated the migration to simply add 3 many to many's on the outreach model (which subsequently make the through tables that the ticket was asking for).

**Comment by darrint** [comment]
+1

File: mypartners/models.py [hunk 2]
```diff
@@ -896,11 +896,22 @@ def __unicode__(self):
...
+    def __unicode__(self):
+        return self.state
+
 
 class NonUserOutreach(models.Model):
     date_added = models.DateTimeField(auto_now_add=True)
     outreach_email = models.ForeignKey("OutreachEmailAddress")
-    from_email = models.EmailField(max_length=255, verbose_name="Email", 
-                                   help_text="Email outreach effort sent from.")
+    from_email = models.EmailField(
```
Comment (tdphillips) [inline mypartners/models.py]: Do we also need to track whatever's in "to" or "cc"?

Comment (JLMcLaugh) [inline mypartners/models.py]: +1, I don't know how we'd be able to tie it to a contact or suggest a partner otherwise.

Comment (aspidites) [inline mypartners/models.py]: outreach_email IS the to email address (not that it's a foreign key to an outreach email address). CC is intentionally being ignored, per conversation by @haliscooper

**Comment by tdphillips** [comment]
There's now a duplicate mypartners migration.

**Comment by aspidites** [comment]
@tdphillips, fixed.

**Comment by darrint** [comment]
:+1:

**Pull request merged**